acceptance/api_v1_services_mutation_test.go

```diff
@@ -0,0 +1,787 @@
+						serverURL, org, service),
+					strings.NewReader(`{ "unbind" : false }`))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+
+				defer response.Body.Close()
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(response.StatusCode).To(Equal(http.StatusOK), string(bodyBytes))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(string(bodyBytes)).To(Equal("{\"boundapps\":null}"))
+			})
+		})
+	})
+
+	Describe("POST api/v1/orgs/:org/applications/:arg/services/", func() {
+		var app string
+
+		BeforeEach(func() {
+			app = newAppName()
+		})
+
+		It("returns a 'bad request' for a non JSON body", func() {
+			response, err := Curl("POST",
+				fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s/services",
+					serverURL, org, app),
+				strings.NewReader(``))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("unexpected end of JSON input\n"))
+		})
+
+		It("returns a 'bad request' for a non-object JSON body", func() {
+			response, err := Curl("POST",
+				fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s/services",
+					serverURL, org, app),
+				strings.NewReader(`[]`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("json: cannot unmarshal array into Go value of type models.BindRequest\n"))
+		})
+
+		It("returns a 'bad request' for JSON object without `name` key", func() {
+			response, err := Curl("POST",
+				fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s/services", serverURL, org, app),
+				strings.NewReader(`{}`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("Cannot bind service without a name\n"))
+		})
+
+		It("returns a 'not found' when the org does not exist", func() {
+			response, err := Curl("POST",
+				fmt.Sprintf("%s/api/v1/orgs/bogus/applications/_dummy_/services", serverURL),
+				strings.NewReader(`{ "name": "meh" }`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("Organization 'bogus' does not exist\n"))
+		})
```

Comment:
a test is missing to check what happens when the org exists but the application doesn't.

Reply:
Will fix.

Reply:
Commit 45223beed41f7017fdc69428d33cdfdf3abc97a3